test/test_payu.py

```diff
@@ -301,3 +301,134 @@ def test_list_archive_dirs():
+    # Create a simple python script
+    python_script = tmp_path / 'test_script.py'
+    with open(python_script, 'w') as f:
+        f.writelines([
+            f"with open('{tmp_path}/output.txt', 'w') as f:\n",
+            "   f.write('Test Python user script')"
+        ])
+
+    # Test run userscript
+    payu.fsops.run_script_command('test_script.py', tmp_path)
+
+    # Check script output
+    with open((tmp_path / 'output.txt'), 'r') as f:
+        assert f.read() == "Test Python user script"
+
+
+def test_run_userscript_python_script_with_shebang_header(tmp_path):
+    # Create a simple python script
+    python_script = tmp_path / 'test_script_with_python_shebang.py'
+    with open(python_script, 'w') as f:
+        f.writelines([
+            '#!/usr/bin/env python\n'
+            f"with open('{tmp_path}/output.txt', 'w') as f:\n",
+            "   f.write('Test Python user script with python shebang header')"
+        ])
+
+    # Make file executable
+    os.chmod(python_script, 0o775)
+
+    # Test run userscript
+    payu.fsops.run_script_command('test_script_with_python_shebang.py',
+                                  tmp_path)
+
+    # Check script output
+    with open((tmp_path / 'output.txt'), 'r') as f:
+        assert f.read() == "Test Python user script with python shebang header"
+
+
+def test_run_userscript_bash_script(tmp_path):
+    # Create a simple bash script
+    bash_script = tmp_path / 'test_script.sh'
+    with open(bash_script, 'w') as f:
+        f.writelines([
+            '#!/bin/bash\n',
+            'echo -n "Test bash user script" > output.txt'
+        ])
+
+    # Test execute script
+    with cd(tmp_path):
+        payu.fsops.run_script_command('./test_script.sh', tmp_path)
+
+    # Check script output
+    with open((tmp_path / 'output.txt'), 'r') as f:
+        assert f.read() == "Test bash user script"
+
+
+def test_userscript_unknown_extension(tmp_path):
+    # Create a text file
+    text_file = tmp_path / 'test_txt.txt'
+    text_file.touch()
+
+    # Test user script raises an error
+    with pytest.raises(RuntimeError):
+        payu.fsops.run_script_command(str(text_file), tmp_path)
+
+
+def test_userscript_non_existent_file(tmp_path):
+    # Test user script raises an error
+    with pytest.raises(RuntimeError):
+        payu.fsops.run_script_command('unknown_userscript.sh',
+                                      tmp_path)
+
+
+def test_run_userscript_python_script_eror(tmp_path):
+    # Create a python script that'll exit with an error
+    python_script = tmp_path / 'test_script_error.py'
+    with open(python_script, 'w') as f:
+        f.write('raise ValueError("Test that script exits with error")')
+
+    # Test userscript raises an error
+    with pytest.raises(RuntimeError):
+        payu.fsops.run_script_command('test_script_error.py',
+                                      tmp_path)
+
+
+def test_run_userscript_bash_script_eror(tmp_path):
```

Comment:
`test_run_userscript_bash_script_eror` should be `error`

Reply:
Oops! Have just fixed that